src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/ThemeManager.cs

```diff
@@ -232,6 +227,31 @@ private static Uri GetFluentWindowThemeColorResourceUri(string systemTheme, bool
         return new Uri("pack://application:,,,/PresentationFramework.Fluent;component/Resources/Theme/" + themeColorFileName, UriKind.Absolute);
     }
 
+    private static void FindFluentThemeAndColorDictionary(out ResourceDictionary fluentThemeDictionary, out ResourceDictionary fluentColorDictionary)
+    {
+        fluentThemeDictionary = null;
+        fluentColorDictionary = null;
+
+        if (Application.Current == null) return;
+
+        foreach (ResourceDictionary mergedDictionary in Application.Current.Resources.MergedDictionaries)
```

Comment:
Is there a possibility to optimize this?

Reply:
One option would be to always put the dictionary as first (which probably we should be doing anyway) and then only check the first one or two.

Reply:
I will go ahead, resolve this thread and raise subsequent PRs where I have changed the method entirely. It will be the second PR from here on.